File: moldme/DTOs/CompanyDto.cs
using System.ComponentModel.DataAnnotations;
using moldme.Models;

namespace moldme.DTOs;

/// <summary>
/// Represents a data transfer object for a company.
/// </summary>
public class CompanyDto
{
    /// <summary>
    /// Gets or sets the name of the company.
    /// </summary>
    [Required]
    [StringLength(64)]
    public string Name { get; set; }

    /// <summary>
    /// Gets or sets the tax identification number of the company.
    /// </summary>
    [Required]
    [Range(100000000, 999999999)]
    public int TaxId { get; set; }

    /// <summary>
    /// Gets or sets the address of the company.
    /// </summary>
    [Required]
    [StringLength(64)]
    public string Address { get; set; }

    /// <summary>
    /// Gets or sets the contact number of the company.
    /// </summary>
    [Required]
    [Range(100000000, 999999999)]
    public int Contact { get; set; }

    /// <summary>
    /// Gets or sets the email of the company.
    /// </summary>
    [Required]
    [StringLength(64)]
    public string Email { get; set; }

    /// <summary>
    /// Gets or sets the sector of the company.
    /// </summary>
    [Required]
    [StringLength(64)]
    public string Sector { get; set; }

    /// <summary>
    /// Gets or sets the subscription plan of the company.
    /// </summary>
    [Required]
    public SubscriptionPlan Plan { get; set; }

    /// <summary>
    /// Gets or sets the password for the company account.
    /// </summary>
    [Required]
    [StringLength(256)]
    public string Password { get; set; }
}

File: moldme_frontend/lib/dtos/companyDTO.dart
class CompanyDTO {
  final String name;
  final int taxId;
  final String address;
  final int contact;
  final String email;
  final String sector;
  final int plan;
  final String password;

  CompanyDTO({
    required this.name,
    required this.taxId,
    required this.address,
    required this.contact,
    required this.email,
    required this.sector,
    required this.plan,
    required this.password,
  });

  /// Factory para criar o DTO a partir de um JSON.
factory CompanyDTO.fromJson(Map<String, dynamic> json) {
    return CompanyDTO(
      name: json['name'] ?? 'Unknown Name',
      taxId: json['taxId'] ?? 0,
      address: json['address'] ?? 'Unknown Address',
      contact: json['contact'] ?? 0,
      email: json['email'] ?? 'Unknown Email',
      sector: json['sector'] ?? 'Unknown Sector',
      plan: json['plan'] ?? 0,
      password: json['password'] ?? '',
    );
  }

  /// Método para converter o DTO para JSON.
  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'taxId': taxId,
      'address': address,
      'contact': contact,
      'email': email,
      'sector': sector,
      'plan': plan,
      'password': password,
    };
  }
}
